pkg/jsontabwriter/jsontabwriter.go

```diff
@@ -0,0 +1,161 @@
+package jsontabwriter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/json2table"
+	"github.com/spf13/viper"
+)
+
+var outputFormatErr = "invalid format: %s"
+
+// TODO: remove cols as function parameter once --cols flag fix is ready
+func GenerateOutput(rootPath string, jsonPaths map[string]string, obj interface{}, cols []string) (string, error) {
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutput(rootPath, obj, jsonPaths, cols, false)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateOutputPreconverted(obj interface{}, convertedObj []map[string]interface{}, cols []string) (string, error) {
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutput("", convertedObj, nil, cols, true)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func generateJSONOutput(obj interface{}) (string, error) {
+	out, err := json.MarshalIndent(obj, "", "\t")
+	if err != nil {
+		return "", err
+	}
+
+	return string(out) + "\n", nil
+}
+
+func generateTextOutput(rootPath string, obj interface{}, jsonPaths map[string]string, cols []string, skipConversion bool) (string, error) {
+	var table []map[string]interface{}
+	var err error
+
+	if skipConversion {
+		table = obj.([]map[string]interface{})
+	} else {
+		table, err = json2table.ConvertJSONToTable(rootPath, jsonPaths, obj)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	var buff = new(bytes.Buffer)
+	var w = new(tabwriter.Writer)
+	w.Init(buff, 5, 0, 3, ' ', tabwriter.StripEscape)
+
+	if !viper.IsSet(constants.ArgNoHeaders) {
+		if _, err = fmt.Fprintln(w, strings.Join(cols, "\t")); err != nil {
+			return "", nil
+		}
+	}
+
+	for _, t := range table {
+		format, values := convertTableToText(cols, t)
+		if _, err = fmt.Fprintf(w, format+"\n", values...); err != nil {
+			return "", err
+		}
+	}
+
+	if err = w.Flush(); err != nil {
+		return "", err
+	}
+
+	return buff.String(), nil
+}
```

Comment:
1. Using booleans as argument is really hard to read e.g what does `foo("bar", true)` do differently vs `foo("bar", false)`? I would consider using a separate function, or a Go Option for generating without conversion
2. This func does too much and should probably be split into multiple helper, private funcs

```suggestion
func generateTextOutputWithConversion(rootPath string, obj interface{}, jsonPaths map[string]string, cols []string) (string, error) {
	table, err := convertToTable(rootPath, jsonPaths, obj)
	if err != nil {
		return "", err
	}
	return writeTableToText(table, cols), nil
}

func generateTextOutputWithoutConversion(convertedObj []map[string]interface{}, cols []string) (string, error) {
	return writeTableToText(convertedObj, cols), nil
}

func convertToTable(rootPath string, jsonPaths map[string]string, obj interface{}) ([]map[string]interface{}, error) {
	return json2table.ConvertJSONToTable(rootPath, jsonPaths, obj)
}

func writeTableToText(table []map[string]interface{}, cols []string) string {
	var buff = new(bytes.Buffer)
	var w = new(tabwriter.Writer)
	w.Init(buff, 5, 0, 3, ' ', tabwriter.StripEscape)

	if !viper.IsSet(constants.ArgNoHeaders) {
		fmt.Fprintln(w, strings.Join(cols, "\t"))
	}

	for _, t := range table {
		format, values := convertTableToText(cols, t)
		fmt.Fprintf(w, format+"\n", values...)
	}

	w.Flush()
	return buff.String()
}
```